test/test_ops.py

```diff
@@ -165,6 +166,34 @@ def test_numpy_ref(self, device, dtype, op):
         finally:
             torch.set_default_dtype(cur_default)
 
+    # Tests that the cpu and gpu results are consistent
+    @onlyCUDA
+    @suppress_warnings
+    @ops(_ops_and_refs_with_no_numpy_ref, dtypes=OpDTypes.any_one)
+    def test_compare_cpu(self, device, dtype, op):
+
+        if dtype not in op.dtypes:
+            raise unittest.SkipTest("This test requires both cuda and cpu support")
+        if dtype not in op.dtypesIfCUDA:
+            raise unittest.SkipTest("This test requires both cuda and cpu support")
+
+        def to_cpu(arg):
+            if isinstance(arg, torch.Tensor):
+                return arg.to(device='cpu')
+            return arg
+
+        samples = op.sample_inputs(device, dtype)
+
+        for sample in samples:
+            cpu_sample = sample.transform(to_cpu)
+            cuda_results = op(sample.input, *sample.args, **sample.kwargs)
+            cpu_results = op(cpu_sample.input, *cpu_sample.args, **cpu_sample.kwargs)
+
+            cuda_results = sample.output_process_fn_grad(cuda_results)
```

Comment:
This is going to need a comment -- why are the results not compared directly?

Reply:
`output_process_fn_grad` has a very unfortunate name. It should be `make_output_comparable` or smth lol
We use this function in `linalg` extensively to postprocess the inputs of functions that are not completely well-defined. Think svd and muliplying the singular vectors by `-1`. CPU and CUDA implementations of the SVD can return valid SVDs that are different. We use this function to compare them.

Reply:
comment added